src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Helpers.cs

```diff
@@ -26,9 +26,11 @@ private void ValidateWritingValue()
                 else
                 {
                     Debug.Assert(_tokenType != JsonTokenType.PropertyName);
-                    if (!_isNotPrimitive && _tokenType != JsonTokenType.None)
+
+                    // It is more likely for CurrentDepth to not equal 0 when writing valid JSON, so check that first to rely on short-circuiting and return quickly.
+                    if (_tokenType != JsonTokenType.None && CurrentDepth == 0)
```

Comment:
I think these comparisons should be reversed according to the comment. It's in the correct order in the other part of the PR.

Reply:
Good catch. That was the intention.